chaincode/fabcar/external/fabcar.go

```diff
@@ -115,7 +115,7 @@ func (s *SmartContract) QueryCar(ctx contractapi.TransactionContextInterface, ca
 // QueryAllCars returns all cars found in world state
 func (s *SmartContract) QueryAllCars(ctx contractapi.TransactionContextInterface) ([]QueryResult, error) {
 	startKey := "CAR0"
-	endKey := "CAR99"
+	endKey := "CAR999"
```

Comment:
Sorry for the late review, but actually the startKey and endKey approach in Fabcar has always been wrong.
There is no need to create artificial startKey and endKey like this at all.
To do an open-ended query, empty string should be used for both startKey and endKey. This approach entirely avoids the problem of trying to decide which keys to use at the extreme ends.
It would be great if we could use the empty string approach in all chaincode languages.

Reply:
Ah, good point. Thanks Dave!

Reply:
> Sorry for the late review, but actually the startKey and endKey approach in Fabcar has always been wrong.
> There is no need to create artificial startKey and endKey like this at all.
> To do an open-ended query, empty string should be used for both startKey and endKey. This approach entirely avoids the problem of trying to decide which keys to use at the extreme ends.
> It would be great if we could use the empty string approach in all chaincode languages.

Thanks for review, your suggestion is right, I modify the query condition to empty string already.